Lib/test/test_operator.py

```diff
@@ -508,6 +510,44 @@ def __getitem__(self, other): return 5  # so that C is a sequence
         self.assertEqual(operator.ixor     (c, 5), "ixor")
         self.assertEqual(operator.iconcat  (c, c), "iadd")
 
+    def test_iconcat_without_getitem(self):
+        operator = self.module
+
+        msg = "'int' object can't be concatenated"
+        with self.assertRaisesRegex(TypeError, msg):
+            operator.iconcat(1, 0.5)
+
+    def test_index(self):
```

Comment:
Looking at the code, it seems like this one has some additional behavior in the C version that isn't reflected in the Python version.

1. Returning a subclass of int raises a DeprecationWarning:
```pycon
>>> class X:
...     def __index__(self): return True
... 
>>> operator.index(X())
<stdin>-3:1: DeprecationWarning: __index__ returned non-int (type bool).  The ability to return an instance of a strict subclass of int is deprecated, and may be removed in a future version of Python.
  operator.index(X())
1
```

But the Python version returns True.

2. If a subclass of int overrides `__index__`, the C version ignores it:
```
>>> class X(int):
...     def __index__(self): return 42
... 
>>> index(X())
42
>>> operator.index(X())
0
```

Arguably these are bugs, but I am not sure it's worth complicating the Python implementation for.

Reply:
It is a known bug, and there is an open issue for this.